components/x-interaction/src/concerns/register-component.js

```diff
@@ -1,9 +1,15 @@
 const registeredComponents = {}
 
-export function registerComponent(component) {
-	registeredComponents[component.wrappedDisplayName] = component
+export function registerComponent(component, name) {
+	const xInteractionName = Symbol('x-interaction-name')
```

Comment:
Maybe more of a Javascript question but I've never seen `Symbol` be used before 🙈 Reading up on it looks like it helps create a unique name here? I'm a tiny bit confused about what is going on here.

Reply:
I have also not encountered `Symbol` before, and also had a question about its usage.

> Every symbol value returned from `Symbol()` is unique.

> `Symbol('foo') === Symbol('foo')  // false`

(Ref. https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Symbol)

If the benefit of using Symbol is that it is unique, is this perhaps undone by later identifying it by converting it into a string (``key.toString() === `Symbol(x-interaction-name)` ``), i.e. wouldn't all `xInteractionName` consts whose values are `Symbol('x-interaction-name')` all be the same?

Is there a risk from them all being the same or could we just use a string instead?

(I don't know much about `Symbol` so very likely I have missed something here.)

Reply:
This was done as part of the step: 
> remove reliance on component `displayName`/`name` in `registerComponent`, instead have an explicit `name` argument, and store that name as an opaque property on the component itself

Having the name be a Symbol would make it semi-private. So if you did `Object.keys(component)` or `for (key in component)`, the `Symbol(x-interaction-name)` won't show up.

But your point does make me realize that if a component gets registered more than once, it will create multiple Symbols storing the name. I am not sure if the component getting registered multiple times is something that could happen, but it might be worth to add a check if the xInteractionName Symbol already exists. I am also not sure why the name needs to be a private property, @apaleslimghost do you remember why you proposed that on the ticket?

Reply:
i suggested it as a way of hiding an internal implementation detail from users of these components, so they wouldn't be able to use something that might break in the future. it's not necessary though.

i was intending for the `Symbol` to be defined outside of the scope of `registerComponent`, so it would always be the same symbol used for multiple component's keys. that would mean you don't have to do the `toString` later.

Reply:
ahh!! that makes much more sense! I'll change it.. thanks Kara!

Reply:
oh, another thought, it would probably be a good idea to throw an error, or at least log a warning, if multiple components are registered with the same name. there's no reason to do this intentionally, and it might lead to the wrong component, or a different version of the same component, being used accidentally, which could break things